spark-jobs/src/test/scala/za/co/absa/enceladus/conformance/streaming/HyperConformanceIntegrationSuite.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2018 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.conformance.streaming
+
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema
+import org.scalatest.FunSuite
+import za.co.absa.enceladus.conformance.interpreter.fixtures.{NestedStructsFixture, StreamingFixture}
+
+class HyperConformanceIntegrationSuite extends FunSuite with StreamingFixture with NestedStructsFixture {
+
+  test("Test with catalyst workaround") {
+    val frame: DataFrame = testHyperConformance(standardizedDf,
+      "result",
+      nestedStructsDS)
+    frame.show()
+    val fields = frame.schema.fieldNames
+    val data = frame.first().getValuesMap(fields)
```

Comment:
Rather than comparing specific values, we can compare full output by doing:
```
df.toJSON.collect().mkString("\n")
```
and comparing results with the expected JSON somewhere in resources. If there. are non-deterministic columns, we can remove them from comparison by using `dropColumn()` or `select()`.

I'd prefer doing `select` on all fields expected in the output data frame. Since changes of column order are allowed, doing `select` will make th order specific for easier comparison. 

Ideally, the expected output JSON should be the same for batch and streaming jobs. If the order of records could be changes, I'd recommend doing `orderBy` before comparison.

Reply:
Makes sense